components/supervisor-api/terminal.proto

```diff
@@ -47,36 +54,56 @@ service TerminalService {
     rpc SetSize(SetTerminalSizeRequest) returns (SetTerminalSizeResponse) {}
 }
 
+message TerminalSize {
+    uint32 rows = 1;
+    uint32 cols = 2;
+    uint32 widthPx = 3;
+    uint32 heightPx = 4;
+}
+
 message OpenTerminalRequest {
     string workdir = 1;
     map<string, string> env = 2;
     map<string, string> annotations = 3;
+
+    string shell = 4;
+    repeated string shell_args = 5;
+
+    TerminalSize size = 6;
 }
 message OpenTerminalResponse {
```

Comment:
At this point it might make sense to return
```proto
message OpenTerminalResponse {
    Terminal terminal = 1;
    string starter_token = 2;
}
```

we basically `get` the terminal we just created